src/main/java/net/imglib2/algorithm/hough/HoughTransforms.java

```diff
@@ -0,0 +1,343 @@
+import net.imglib2.algorithm.localextrema.LocalExtrema;
+import net.imglib2.algorithm.localextrema.LocalExtrema.MaximumCheck;
+import net.imglib2.type.NativeType;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.util.Util;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.Views;
+
+/**
+ * This abstract class provides some basic functionality for use with arbitrary
+ * Hough-like transforms.
+ *
+ * @param <T>
+ *            the data type of the input image.
+ * @author Larry Lindsey
+ * @author Yili Zhao
+ * @author Gabe Selzer
+ */
+public class HoughTransforms< T extends RealType< T > & Comparable< T > >
+{
+
+	public static final int DEFAULT_THETA = 180;
+
+	/**
+	 * Calculates the geometric distance between [0, ... , 0] and position.
+	 * 
+	 * @param position
+	 *            - the position of the second point
+	 * @return {@code float} - the distance between the two points.
+	 */
+	final private static float computeLength( final long[] position )
+	{
+		float dist = 0;
+
+		for ( int d = 0; d < position.length; ++d )
+		{
+			final long pos = position[ d ];
+
+			dist += pos * pos;
+		}
+
+		return ( float ) Math.sqrt( dist );
+	}
+
+	/**
+	 * Calculates a default number of rho bins, which corresponds to a
+	 * resolution of one pixel.
+	 *
+	 * @param inputImage
+	 *            the {@link RandomAccessibleInterval} in question.
+	 * @return default number of rho bins.
+	 */
+	private static int defaultRho( final RandomAccessibleInterval< ? > inputImage )
+	{
+		final long[] dims = new long[ inputImage.numDimensions() ];
+		inputImage.dimensions( dims );
+		return ( int ) ( 2 * computeLength( dims ) );
+	}
+
+	/**
+	 * Returns the size of the vote space output image given an input
+	 * {@link RandomAccessibleInterval}.
+	 * 
+	 * @param input
+	 *            - the {@link RandomAccessibleInterval} over which the Hough
+	 *            Line Transform will be run
+	 * @return {@code long[]} - the dimensions of the vote space image
+	 */
+	public static < T extends RealType< T > > long[] getVotespaceSize( RandomAccessibleInterval< T > input )
+	{
+		return new long[] { defaultRho( input ), DEFAULT_THETA };
+	}
+
+	/**
+	 * Returns the size of the voteSpace output image given desired {@code nRho}
+	 * and {@code nTheta} values.
+	 * 
+	 * @param nRho
+	 *            - the number of bins for rho resolution
+	 * @param nTheta
+	 *            - the number of bins for theta resolution
+	 * @return {@code long[]} - the dimensions of the vote space image.
+	 */
+	public static long[] getVotespaceSize( int nRho, int nTheta )
+	{
+		return new long[] { nRho, nTheta };
+	}
+
+	/**
+	 * Pick vote space peaks with a {@link LocalExtrema}.
+	 *
+	 * @return {@code List<Point>} - a list of all of the local maxima of the
+	 *         {@code voteSpace}.
+	 */
+	public static < T extends RealType< T > & NativeType< T > > List< Point > pickPeaks( RandomAccessibleInterval< T > voteSpace, T minPeak )
```

Comment:
Line is too long, break it like this:
```java
public static < T extends Comparable< T > > List< Point > pickPeaks(
			final RandomAccessibleInterval< T > voteSpace,
			final T minPeak )
```

Reply:
@hanslovsky My imglib2 style sheet is not letting me do this; if I attempt to break this as above and then format the signatures return to one long line. Is there an update I am missing?

Reply:
Other than the first 3 my save actions match. I am also using the same style-sheet as above.

Reply:
I have experienced weird formatting behavior with eclipse, so that might just be it. My eclipse version is:
```
Version: Oxygen.3a Release (4.7.3a)
Build id: 20180405-1200
```
What happens if you format it, save, and after save actions were applied, you hit ctrl+z?

Reply:
Strangely enough I tried it and although it is not automatically formatting the lines it does not revert them to one long line when I format again after breaking the lines. So I guess this works.